lms/djangoapps/ccx/tests/test_field_override_performance.py

```diff
@@ -0,0 +1,172 @@
+"""
+import ddt
+import mock
+from nose.plugins.attrib import attr
+
+from courseware.views import progress  # pylint: disable=import-error
+from django.core import cache
+from django.test.client import RequestFactory
+from django.test.utils import override_settings
+from edxmako.middleware import MakoMiddleware  # pylint: disable=import-error
+from student.models import CourseEnrollment  # pylint: disable=import-error
+from student.tests.factories import UserFactory  # pylint: disable=import-error
+from xmodule.modulestore.tests.django_utils import ModuleStoreTestCase, \
+    TEST_DATA_XML_MODULESTORE, TEST_DATA_SPLIT_MODULESTORE
+from xmodule.modulestore.tests.factories import check_mongo_calls
+from xmodule.modulestore.xml_importer import import_course_from_xml
+from xmodule.tests import DATA_DIR  # pylint: disable=import-error
+
+
+@attr('shard_1')
+@mock.patch.dict(
+    'django.conf.settings.FEATURES', {'ENABLE_XBLOCK_VIEW_ENDPOINT': True}
+)
+@ddt.ddt
+class FieldOverridePerformanceTestCase(ModuleStoreTestCase):
+    """
+    Base class for instrumenting SQL queries and Mongo reads for field override
+    providers.
+    """
+    def setUp(self):
+        """
+        Create a test client, course, and user.
+        """
+        super(FieldOverridePerformanceTestCase, self).setUp()
+
+        self.request_factory = RequestFactory()
+        self.student = UserFactory.create()
+        self.request = self.request_factory.get("foo")
+        self.request.user = self.student
+
+        MakoMiddleware().process_request(self.request)
+
+        # TEST_DATA must be overridden by subclasses, otherwise the test is
+        # skipped.
+        self.TEST_DATA = None
+
+    def setup_course(self, course_name):
+        """
+        Imports some XML course data.
+        """
+        course = import_course_from_xml(
+            self.store,
+            999,
+            DATA_DIR,
+            ['test_increasing_size/graded_{}'.format(course_name)]
+        )[0]
+
+        CourseEnrollment.enroll(self.student, course.id)
+
+        return course
+
+    def grade_course(self, course):
+        """
+        Renders the progress page for the given course.
+        """
+        return progress(
+            self.request,
+            course_id=course.id.to_deprecated_string(),
+            student_id=self.student.id
+        )
+
+    def instrument_course_progress_render(self, course_name, queries, reads):
+        """
+        Renders the progress page, instrumenting Mongo reads and SQL queries.
+        """
+        course = self.setup_course(course_name)
+
+        # Disable the cache
+        # TODO: remove once django cache is disabled in tests
+        single_thread_cache = cache.get_cache(
+            backend='django.core.cache.backends.dummy.DummyCache',
+            LOCATION='single_thread_local_cache'
+        )
+        single_thread_cache.clear()
+        with mock.patch("django_comment_client.permissions.CACHE", single_thread_cache):
```

Comment:
This is probably not the cache that you need to override. You'll have to figure out which caches having data stored to them, and override those cache imports.

Also, given that the cache that you're subbing in is the `DummyCache`, I don't think you actually need to clear it, since the `DummyCache` never actually caches any data.